classes/pointmarking_form.php

```diff
@@ -0,0 +1,139 @@
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Form to mark an observation point
+ *
+ * @package   mod_observation
+ * @copyright  2021 Endurer Solutions Team
+ * @author Matthew Hilton <mj.hilton@outlook.com>
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace mod_observation;
+
+defined('MOODLE_INTERNAL') || die;
+require_once($CFG->libdir.'/formslib.php');
+
+/**
+ * Creates a moodle_form to mark an observation point.
+ *
+ * @package   mod_observation
+ * @copyright  2021 Endurer Solutions Team
+ * @author Matthew Hilton <mj.hilton@outlook.com>
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class pointmarking_form extends \moodleform {
+    /**
+     * Defines the point marking form
+     */
+    public function definition() {
+        $mform = $this->_form;
+
+        $prefill = $this->_customdata;
+
+        $mform->addElement('header', 'obpointheader', get_string('observationpoint', 'observation'));
+
+        // Title.
+        $mform->addElement('static', 'title', get_string('title', 'observation'));
+
+        // Observation point information.
+        $instext = $prefill['ins'];
+        $insformat = $prefill['ins_f'];
+
+        $mform->addElement('static', 'instructions', get_string('gradinginstructions', 'observation'),
+            format_text($instext, $insformat));
+
+        switch($prefill['res_type']) {
+            // Text input type.
+            case 0:
+                $mform->addElement('textarea', 'response', get_string('textinputtype', 'observation'),
+                    ['rows' => 3, 'cols' => 100]);
+                $mform->setType('response', PARAM_RAW);
+                $mform->addRule('response', get_string('required', 'observation'), 'required', null, 'client');
+        }
+
+        $mform->addElement('static', 'max_grade_display', get_string('maxgrade', 'observation'), $prefill['max_grade']);
+
+        $mform->addElement('text', 'grade_given', get_string('gradegiven', 'observation'));
+        $mform->setType('grade_given', PARAM_INT);
+        $mform->addRule('grade_given', get_string('err_numeric', 'form'), 'numeric', null, 'client');
+        $mform->addRule('grade_given', get_string('required', 'observation'), 'required', null, 'client');
+        $mform->addRule('grade_given', get_string('intgreaterthanorzero', 'observation'), 'regex', '/^[0-9]\d*$/', 'client');
+
+        // Extra comment block.
+        $mform->addElement('textarea', 'ex_comment', get_string('extracomment', 'observation'), ['rows' => 3, 'cols' => 100]);
+        $mform->setType('ex_comment', PARAM_RAW);
```

Comment:
can this be PARAM_TEXT? It clears things like naughty `<script>` tags that someone might try to insert. PARAM_RAW should generally be the last option, when you need to do stuff like raw HTML editing etc